operations/rotate-linux-windows-passwords/README.md

```diff
@@ -0,0 +1,102 @@
+# Simple Password Rotation with HashiCorp Vault
+This guide demonstrates an automated password rotation workflow using Vault and a simple Bash script. These scripts could be run in a cron job or scheduled task to dynamically update local system passwords on a regular basis.
+
+NOTE: This is *not* the be-all and end-all of password rotation. It is also not a PAM tool. It can do the following:
+
+* Rotate local system passwords on a regular basis
+* Allow systems to rotate their own passwords
+* Store login credentials securely in Vault
+* Ensure that passwords meet complexity requirements
+* Require users to check credentials out of Vault 
+
+In this demo you'll have three terminal windows open. One for running Vault, a second for entering Vault CLI commands, and one for your remote Linux server. The scripts run from the remote GCP instance and connect to Vault to update their stored credentials.
+
+## Prerequisites
+* Vault installed on your local machine: https://www.vaultproject.io/downloads.html
+* GCP account and SDK command line tools installed: https://cloud.google.com/sdk/
+
+## Estimated Time to Complete
+10 minutes
+
+### Step 1: Configure Vault on your local machine
+In a new terminal, run the following commands to get Vault setup on your laptop:
+
+```
+vault server -dev -dev-root-token-id=root
+```
+
+Leave Vault running in this terminal. You can point out API actions as they are logged, such as revoked leases, etc.
+
+### Step 2: Configure a policy
+Open the Vault UI at http://127.0.0.1:8200 and create an ACL policy called `rotate-linux` with the following content:
+
+```
+path "secret/data/linux/*" {
+  capabilities = ["create", "read", "update", "list"]
+}
+```
+
+### Step 3: Generate a token
+Open a second terminal and generate a token for use in Step 5.
+```
+export VAULT_ADDR=http://127.0.0.1:8200
+vault token create -period 24h -policy rotate-linux
+```
+
+### Step 4: Configure a Linux instance
+Open a third terminal window and run the following commands. The first one creates a new Linux instance in GCP. The second command copies our password rotation script to the remote host. The third establishes an SSH connection to the remote host, while enabling a tunnel back to Vault on our local machine.
+
+```
+gcloud compute instances create linuxdemo \
```

Comment:
Eventually we'd like to be able to utilize this guide for multiple cloud targets. We don't need to do it now for other clouds, but if we could structure the top level guide (README) to be cloud agnostic, and have them run the steps for their specific clouds within subdirectories, we can then re-use the same guide for multiple clouds instead of re-inventing the wheel each time.

So what I think we can do, is create subdirectories within this directory for each cloud, starting with `gcp`, then within those folders have commands for each steps scripted.

For example...

`gcp/configure-linux.sh` would include the 3 `gcloud` commands for step 4 in a script. Then in the top level README you would say something like "In your selected cloud's folder, run the `configure-linux.sh` command. In addition, we can just include the copy/pasteable cloud specific commands in README within (`gcp/README.md`) and link to those from the top level README.

Let me know if that doesn't make sense, could also be an iteration after the initial merge.

Reply:
Why do we need multiple cloud targets?  The principle of the thing is the same.  I really intended this as a quick-and-easy demo for SEs and partners to demonstrate a feature, not a multi-cloud password management solution.

The code itself can run on any linux or windows server, no matter what cloud they are on.

If you want an end-user guide for password rotation i need to rewrite this entirely, remove all cloud references, and just assume that the user has:

1.  A linux or windows instance on *any* cloud
2.  A vault server running somewhere the instance can get to it

That way we don't have to talk about cloud at all. It's simply an interaction between a windows/linux server and a vault server.  You could also run in on premises.

Also as an aside, we really ought to have a separate set of guides that are demo-oriented, because we do things in demos that you'd *never* want to do in QA or production.  Those should be separate from tutorials that teach best practices and basic skills.